client/src/app/callback/callback.component.ts

```diff
@@ -8,7 +12,49 @@ import { Component, OnInit, ViewEncapsulation } from '@angular/core';
 })
 export class CallbackComponent implements OnInit {
 
-  constructor() { }
+  constructor(public auth: AuthService,
+              public jwtHelper: JwtHelperService,
+              private alertService: AlertService,
+              public router: Router) {
+    this.handleAuthentication()
+  }
+
+  public handleAuthentication(): void {
+    let params = this.getJsonFromUrl();
+    const accessToken = params['access_token'];
+    let decodedToken = this.jwtHelper.decodeToken(accessToken);
+
+    if(decodedToken['https://auth.data.humancellatlas.org/group'] != "hca"){
```

Comment:
Is this check still needed?

Reply:
yes, we want to restrict user emails to only be within hca group.